src/cpp/fastdds/xtypes/serializers/idl/dynamic_type_idl.cpp

```diff
@@ -935,8 +1079,12 @@ ReturnCode_t struct_to_idl(
         }
     }
 
-    // Add types name
-    idl << "struct " << node.info.type_kind_name;
+    for (int i = 0; i < n_modules; i++)
+    {
+        idl << TAB_SEPARATOR;
+    }
```

Comment:
This should only be done if there was an extensibility right? Please check if there are any other similar cases in the rest of serializers.

Reply:
A solution could be to add the appendable tag as well, I have tested that Foxglove is able to parse it (maybe ignores it, but at least doesn't fail).